Tools/CISupport/ews-build/steps.py

```diff
@@ -565,6 +570,51 @@ def evaluateCommand(self, cmd):
         return rc
 
 
+class CheckOutPullRequest(steps.ShellSequence):
+    name = 'checkout-pull-request'
+    description = ['checking-out-pull-request']
+    descriptionDone = ['Checked out pull request']
+    haltOnFailure = True
+
+    def __init__(self, **kwargs):
+        super(CheckOutPullRequest, self).__init__(timeout=10 * 60, logEnviron=False, **kwargs)
+
+    def doStepIf(self, step):
+        return self.getProperty('github.number', False)
+
+    def hideStepIf(self, results, step):
+        return not self.doStepIf(None)
+
+    @defer.inlineCallbacks
+    def run(self):
+        self.commands = []
+
+        remote = self.getProperty('github.head.repo.full_name', 'origin').split('/')[0]
+        project = self.getProperty('github.head.repo.full_name', self.getProperty('project'))
+        branch = self.getProperty('github.head.ref', 'main')
+
+        for command in [
```

Comment:
Can consider combining commands from clean-and-update-working-directory step here, and skipping that step in PR workflow, but that can be done later as well. Please atleast include a FIXME for removing that step later-on.